sql-plugin/src/main/301db/scala/com/nvidia/spark/rapids/shims/v2/GpuRegExpReplaceExec.scala

```diff
@@ -62,7 +59,8 @@ class GpuRegExpReplaceMeta(
     if (GpuOverrides.isSupportedStringReplacePattern(expr.regexp)) {
       GpuStringReplace(lhs, regexp, rep)
     } else {
-      GpuRegExpReplace(lhs, regexp, rep)
+      GpuRegExpReplace(lhs, regexp, rep, pattern.getOrElse(
+        throw new SQLException("Expression has not been tagged with cuDF regex pattern")))
```

Comment:
Curious why this is using SQLException?  We've traditionally used IllegalStateException since this is more of a logic error in the meta/override code if it occurs.